activefires_pp/tests/test_fire_detection_id_handling.py

```diff
@@ -205,8 +192,45 @@ def test_add_unique_day_id_to_detections_24hours_plus(readdata, setup_comm, geth
 @patch('socket.gethostname')
 @patch('activefires_pp.post_processing.ActiveFiresPostprocessing._setup_and_start_communication')
 @patch('activefires_pp.post_processing._read_data')
-def test_add_unique_day_id_to_detections_newday(readdata, setup_comm, gethostname,
-                                                fake_yamlconfig_file_post_processing):
+def test_add_unique_day_id_to_detections_newday_from_cache(readdata, setup_comm, gethostname,
+                                                           fake_yamlconfig_file_post_processing_with_id_cache):
+    """Test adding unique id's to the fire detection data."""
+    gethostname.return_value = "my.host.name"
+
+    myborders_file = "/my/shape/file/with/country/borders"
+    mymask_file = "/my/shape/file/with/polygons/to/filter/out"
+
+    afpp = ActiveFiresPostprocessing(fake_yamlconfig_file_post_processing_with_id_cache,
+                                     myborders_file, mymask_file)
+
+    myfilepath = TEST_ACTIVE_FIRES_FILEPATH4
+
+    fstream = io.StringIO(TEST_ACTIVE_FIRES_FILE_DATA4)
+    afdata = pd.read_csv(fstream, index_col=None, header=None, comment='#', names=COL_NAMES)
+    readdata.return_value = afdata
+
+    this = ActiveFiresShapefileFiltering(filepath=myfilepath, timezone='GMT')
+    with patch('os.path.exists') as mypatch:
+        mypatch.return_value = True
+        afdata = this.get_af_data(filepattern=MY_FILE_PATTERN, localtime=False)
+
+    TestCase().assertDictEqual(afpp._fire_detection_id, {'date': datetime(2023, 5, 1, 0, 0),
+                                                         'counter': 1})
+    # 2 new fire detections, so (current) ID should be raised - a new day, so id
+    # starting over from 0, and a new date!
+    afdata = afpp.add_unique_day_id(afdata)
+    assert 'detection_id' in afdata
+    assert afdata['detection_id'].values.tolist() == ['20230618-1', '20230618-2']
+    TestCase().assertDictEqual(afpp._fire_detection_id, {'date': datetime(2023, 6, 18, 9, 56, 0),
+                                                         'counter': 2})
```

Comment:
With recent pythons, you should be able to just do
```suggestion
    assert afpp._fire_detection_id == {'date': datetime(2023, 6, 18, 9, 56, 0), 'counter': 2})
```

Reply:
Ahh, nice, thanks, didn't know that.